app/jobs/preserve_children_job.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+class PreserveChildrenJob < ApplicationJob
+  delegate :metadata_adapter, to: :change_set_persister
+  delegate :query_service, to: :metadata_adapter
+
+  def perform(id:)
+    resource = query_service.find_by(id: id)
```

Comment:
This is true. I don't think we want to pass the member ids, in case there are like a thousand. A member query that takes an ID is a good idea, but seems like a Valkyrie issue yeah?

Reply:
Yes, I'm fine sticking with the current approach here, and I've added a Valkyrie ticket for the member query that takes an ID instead of a resource: https://github.com/samvera/valkyrie/issues/703